### PR TITLE
fix: Set MPID before open session

### PR DESCRIPTION
**Pull request by SbDove**
## Summary
 - checks forwarder settings to see if the identity type is set to MPID on initializations.
 - checks if MPID has already been established on kit initialization. 
 - if both checks are successful, calls `changeUser` via `onUserIdentified` before `openSession`

 ## Testing Plan
- Testing was done by enabling local overrides to test kit initialization with the proper settings - no issues arose from the initialization process. but confirmation that the duplication sessions were corrected was not done. 

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://mparticle.lightning.force.com/lightning/r/Gap_Library__c/a0J3r00000ks1UeEAI/view (issue with duplicate sessions in braze related to the order in which we call `openSession` and `changeUser`)

### CODE REVIEW

**Comment by SbDove** [comment]
@alexs-mparticl

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -755,6 +755,10 @@ var constructor = function () {
                 }
             }
 
+            if (forwarderSettings.userIdentificationType === 'MPID' && mParticle.Identity.getCurrentuser().getMPID() != null) {
+                onUserIdentified(mParitcle.Identity.getCurrentuser())
```
Comment (rk-mp) [inline src/BrazeKit-dev.js:759]: typo in mParticle?

File: src/BrazeKit-dev.js [hunk 2]
```diff
@@ -755,6 +755,10 @@ var constructor = function () {
                 }
             }
 
+            if (forwarderSettings.userIdentificationType === 'MPID' && mParticle.Identity.getCurrentuser().getMPID() != null) {
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:758]: I think you should err on the side of caution and check if both `Identity` and `getCurrentUser` are also not undefined.

**Comment by SbDove** [comment]
@alexs-mparticle Pushed the tests. Let me know if there's anything else that needs to be addressed on this.

File: dist/BrazeKit.common.js [hunk 3]
```diff
@@ -9172,7 +9172,7 @@ window.braze = require$$0;
 var name = 'Appboy',
     suffix = 'v4',
     moduleId = 28,
-    version = '4.1.0',
+    version = '4.1.1',
```
Comment (alexs-mparticle) [inline dist/BrazeKit.common.js:9175]: Please undo this change. Our process is to update the version number and distribution as a separate commit when we release.

File: dist/BrazeKit.iife.js [hunk 4]
```diff
@@ -9172,7 +9172,7 @@ var mpBrazeKitV4 = (function (exports) {
 	var name = 'Appboy',
 	    suffix = 'v4',
 	    moduleId = 28,
-	    version = '4.1.0',
+	    version = '4.1.1',
```
Comment (alexs-mparticle) [inline dist/BrazeKit.iife.js:9175]: Please undo all changes in this file as well.

File: package-lock.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
     "packages": {
         "": {
             "name": "@mparticle/web-braze-kit",
-            "version": "4.0.1",
+            "version": "4.1.1",
```
Comment (alexs-mparticle) [inline package-lock.json:9]: Please roll this back.

File: src/BrazeKit-dev.js [hunk 6]
```diff
@@ -760,7 +760,7 @@ var constructor = function () {
                 finishBrazeInitialization(forwarderSettings);
             } else {
                 if (!braze.initialize(forwarderSettings.apiKey, options)) {
-                    return 'Failed to initialize: ' + name;
+                    return 'Failed to initialize: ' + name;corrects
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:763]: Why is this `corrects;` statement just dangling here?

File: test/tests.js [hunk 7]
```diff
@@ -244,10 +244,26 @@ describe('Braze Forwarder', function() {
...
         },
         reportService = new ReportingService();
-
+        mParticle.Identity = {
+            getCurrentUser: function () {
+                return {
+                    getMPID: function () {
+                        return 'MPID123';
+                    },
+                    getUserIdentities: function () {
```
Comment (alexs-mparticle) [inline test/tests.js:253]: is `getUserIdentities` a necessary addition?

File: test/tests.js [hunk 8]
```diff
@@ -244,10 +244,26 @@ describe('Braze Forwarder', function() {
...
+                            },
+                        };
+                    },
+                };
+            },
+        };
     before(function() {
         // expandCommerceEvent is tightly coupled to mParticle being loaded
         // as well as having a few parameters on the Store.
+        // debugger;
```
Comment (alexs-mparticle) [inline test/tests.js:266]: Please remove this debugger comment.

File: test/tests.js [hunk 9]
```diff
@@ -2065,5 +2091,7 @@ USD,
 
             impressionEvent.should.eql(expectedImpressionEvent);
         });
+
+
```
Comment (alexs-mparticle) [inline test/tests.js:2095]: You can remove these extra spaces.

```suggestion
```

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]
The changes look good. Please remove the changes to dist and and package-lock.json. These will be handled as part of our release job.

File: dist/BrazeKit.common.js (hunk 3 above)
Comment (SbDove) [reply dist/BrazeKit.common.js:9175]: This change seems to already be [in the master-v4 branch](https://github.com/mparticle-integrations/mparticle-javascript-integration-braze/blob/b043d4637a7e0b6d82bf550d63936a77d6b2e0c2/dist/BrazeKit.common.js#L9175). I don't think it was a change that I made. Not sure I know how to roll it back in this case.

File: dist/BrazeKit.iife.js (hunk 4 above)
Comment (SbDove) [reply dist/BrazeKit.iife.js:9175]: This change is also already [in the master-v4 branch](https://github.com/mparticle-integrations/mparticle-javascript-integration-braze/blob/b043d4637a7e0b6d82bf550d63936a77d6b2e0c2/dist/BrazeKit.iife.js#L9175).

File: package-lock.json (hunk 5 above)
Comment (SbDove) [reply package-lock.json:9]: Looking at the [git blame](https://github.com/mparticle-integrations/mparticle-javascript-integration-braze/blame/b043d4637a7e0b6d82bf550d63936a77d6b2e0c2/package-lock.json#L9) it says that you made the change in the master-v4 branch a few days ago. I could very well be confused about these changes.

File: test/tests.js (hunk 7 above)
Comment (SbDove) [reply test/tests.js:253]: Currently, it is necessary, yes. The getUserIdentities function is needed in the onUserIdentified call which sets the Braze identity. In retrospect, I'm calling the onUserIdentified function which in turn calls the Braze changeUser function. Seems like I could skip the secondary function and call the Braze function directly - would it make sense to make that change to eliminate the getUserIdentities and the extra processing?

File: test/tests.js (hunk 9 above)
Comment (SbDove) [reply test/tests.js]: done

File: test/tests.js (hunk 8 above)
Comment (SbDove) [reply test/tests.js]: done

File: src/BrazeKit-dev.js (hunk 6 above)
Comment (SbDove) [reply src/BrazeKit-dev.js]: I don't remember. Removed.

File: dist/BrazeKit.common.js (hunk 3 above)
Comment (alexs-mparticle) [reply dist/BrazeKit.common.js:9175]: OK, you may need to rebase your branch against `master-v4`. I'll see if I can help you out with that.

**Review by alexs-mparticle — APPROVED** [review]
